src/controllers/zodiac.ts

```diff
@@ -0,0 +1,40 @@
+import { Request, Response } from "express";
+import { Signs_ZodiacDirBG } from "../models/signs";
+import {
+  GetLastElementJSNatFromSignData,
+  getTodayAllSignRecordsFromDB,
+} from "../dal/zodiacdeep";
+import { ISignRecord } from "../interfaces/signs";
+import { getText } from "../helpers/arrayProcessing";
+
+export async function getZodiac(req: Request, res: Response): Promise<void> {
+  {
+    const sign = req?.query?.sign;
+    let totalResult = "";
+
+    if (typeof sign !== "undefined") {
+      if (Signs_ZodiacDirBG.includes(sign as string)) {
+        let result: Array<ISignRecord> = [];
+        let element = await GetLastElementJSNatFromSignData(sign as string);
+        if (element) {
+          result.push(element);
+        }
+        totalResult = getText(result);
+      } else if ((sign as string).toLowerCase() === "zodiac") {
+        const resultTextArray = await getTodayAllSignRecordsFromDB();
+        if (resultTextArray) {
+          totalResult = getText(resultTextArray);
+        }
+      }
+    } else {
+      const resultTextArray = await getTodayAllSignRecordsFromDB();
+      if (resultTextArray) {
+        totalResult = getText(resultTextArray);
+      }
```

Comment:
NOTE:

I can see that you are doing - getText() after receiving each result. You can abstract it in a separate function OR if the idea of your function is to return the value from getText() all the time you can incorporate it in the other method.